fanficfare/configurable.py

```diff
@@ -691,7 +691,7 @@ def get_config(self, sections, key, default=""):
     # split and strip each.
     def get_config_list(self, sections, key, default=[]):
         vlist = re.split(r'(?<!\\),',self.get_config(sections,key)) # don't split on \,
-        vlist = [x for x in [ v.strip().replace('\,',',') for v in vlist ] if x !='']
+        vlist = [x for x in [ v.strip().replace(r'\,',',') for v in vlist ] if x !='']
```

Comment:
??? Are you sure this change is appropriate?  That's not a regexp replace, it's the literal string `'\,'`  It's an escape by FFF so a `,` can be included in a list value if needed.  (I don't remember _why_ it might be needed, but that's what it does...)

Reply:
Ok, my bad. I tried to fix the warnings but I didn't understand enough what I was doing...

The warning was :
```
configurable.py:694
  ~/src/perso/FanFicFare/fanficfare/configurable.py:694: DeprecationWarning: invalid escape sequence \,
    vlist = [x for x in [ v.strip().replace('\,',',') for v in vlist ] if x !='']
```

Reply:
The change is correct! `r''` strings have nothing to do with regular expressions.

```
$ python -W default
Python 3.8.3 (default, May 17 2020, 18:15:42) 
[GCC 10.1.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> r'\,'
'\\,'
>>> '\,'
<stdin>:1: DeprecationWarning: invalid escape sequence \,
'\\,'
```

The backslash is ambiguous; it *could* mean "give me the character represented by the escape sequence 'comma'", but there is no such escape sequence listed in https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals
> Unlike Standard C, all unrecognized escape sequences are left in the string unchanged, i.e., the backslash is left in the result.

So the failover mode when you add C-style escape sequences that don't exist, is to treat it as a string literal. But this is deprecated behavior, and future versions of python will make it a hard error.

Using `r''` string literals merely disables this ambiguity by ensuring the string is not parsed for C-style escape sequences, which is fine as you don't have any. The python interpreter parses the source code the same way.

Reply:
Saying `r''` has _nothing_ to do with regexp seems excessive; that's basically the only thing I've previously seen it used for.

But whatever, here's a fix for deprecation warnings:
188c92332ffaa614091659b211d77c9220138dcf

(I was tempted to use `R''` just to be contrary, but I resisted.)